### PR TITLE
Feature/connect api

### PR DESCRIPTION
**Pull request by RileyManda**
I have completed the task of connect-API according to the stipulated  [requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/connect_to_api_v2_1.md): 🌱 

**The following implementations were made:** 🌱 
- [x] Made use of the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as my backend.
- [x] Used [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make your HTTP requests to the backend
- [x] Fetched books from the Bookstore API after the component has rendered
- [x] Refactored add book and remove book features to persist changes in the server.

In addition, I have ensured that this project fulfills the following default requirements :
✅ - The application behaves the same way after the refactoring
✅ - There are [no linter errors](https://github.com/microverseinc/linters-config).
✅ - Used correct flow [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
✅ - Documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
✅ - Followed the list of the [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md) and  [JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

Please review my implementation and suggest any improvements and changes.

Thank you 🙏

### CODE REVIEW

File: src/components/AddBookButton.js [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
     const newBook = {
-      item_id: id,
+      item_id: uuidv4(),
       title: bookTitle,
       author: bookAuthor,
+      category: bookCategory,
     };
 
     dispatch(addBook(newBook));
   };
```
Comment (ShadyShawkat) [inline src/components/AddBookButton.js]: - [ ] Kindly fix the following behavior. Currently, this is what happens when the user tries to add or remove a book. (See attached GIF)

chrome-capture-2023-6-26

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @RileyManda, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/AddBookButton.js (hunk 1 above)
Comment (RileyManda) [reply src/components/AddBookButton.js]: Thank you for reviewing my implementation and providing feedback. Looks like I might have broken the dispatch on add and remove functionality and will work on the changes. Thank you

**Comment by RileyManda** [comment]
I have completed the task of fixing the run time bug in the application related to the lack of reactive Ui updates on Add and Remove book as requested by the previous reviewer. ⭐ 

In addition, I have ensured that this project fulfills the following default requirements :

✅ - The application behaves the same way after the refactoring
✅ - There are [no linter errors](https://github.com/microverseinc/linters-config).
✅ - Used correct flow [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
✅ - Documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
✅ - Followed the list of the [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md) and  [JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

Please review my implementation and suggest any improvements and changes.

Thank you 🙏

**Review by shella12 — APPROVED** [review]
Hi @RileyManda, 


## Status : APPROVED 🥳 🥳 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- Fixed adding new book problem ✔️ 
- Implemented all requirements 👏 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**